### PR TITLE
[SDK-400] Validate url, secret, and topic list on webhook creation.

### PR DESCRIPTION
**Pull request by sfendell-labelbox**
Check that these fields are non-empty on creation to match frontend behavior.

### CODE REVIEW

File: labelbox/schema/webhook.py [hunk 1]
```diff
@@ -77,9 +77,15 @@ def create(client, topics, url, secret, project) -> "Webhook":
             ValueError: If the topic is not one of Topic or status is not one of Status
 
         Information on configuring your server can be found here (this is where the url points to and the secret is set).
-                        https://docs.labelbox.com/en/configure-editor/webhooks-setup#setup-steps
+                        https://docs.labelbox.com/reference/webhook
```
Comment (sfendell-labelbox) [inline labelbox/schema/webhook.py:80]: A little out of scope, but this docs link is broken.

Comment (vbrodsky) [reply labelbox/schema/webhook.py:80]: nice!

File: labelbox/schema/webhook.py [hunk 2]
```diff
@@ -77,9 +77,15 @@ def create(client, topics, url, secret, project) -> "Webhook":
...
 
         """
+        if not secret:
+            raise TypeError(f"Secret must be a non-empty string.")
```
Comment (vbrodsky) [inline labelbox/schema/webhook.py:84]: I think `ValueError` is more appropriate here

Comment (sfendell-labelbox) [reply labelbox/schema/webhook.py]: Thanks, fixed.

File: labelbox/schema/webhook.py [hunk 3]
```diff
@@ -77,9 +77,15 @@ def create(client, topics, url, secret, project) -> "Webhook":
             ValueError: If the topic is not one of Topic or status is not one of Status
 
         Information on configuring your server can be found here (this is where the url points to and the secret is set).
-                        https://docs.labelbox.com/en/configure-editor/webhooks-setup#setup-steps
+                        https://docs.labelbox.com/reference/webhook
 
         """
+        if not secret:
+            raise ValueError(f"Secret must be a non-empty string.")
```
Comment (kopreschko) [inline labelbox/schema/webhook.py:84]: There's no need for an f-string here, as there are no variables/expressions in the string. This goes for all the messages below.

File: tests/integration/test_webhook.py [hunk 4]
```diff
@@ -40,3 +40,39 @@ def test_webhook_create_update(project, rand_gen):
         "Topics must be List[Webhook.Topic]. Found `invalid..`"
 
     webhook.delete()
+
+
```
Comment (kopreschko) [inline tests/integration/test_webhook.py:44]: These, in my opinion, should be considered unit-tests.

**Review by kopreschko — COMMENTED** [review]
Nice work in both the API and SDK by adding tests for these validations! Just some minor observations, the rest looks good to me.

File: labelbox/schema/webhook.py (hunk 3 above)
Comment (sfendell-labelbox) [reply labelbox/schema/webhook.py]: Fixed.

File: tests/integration/test_webhook.py (hunk 4 above)
Comment (sfendell-labelbox) [reply tests/integration/test_webhook.py:44]: There's a lot of tests in `integration` that could be better considered a unit test, but this needs a configured project and client which only exist in integration fixtures, and pulling them out would be too much of a pain for this I think.

Comment (kopreschko) [reply tests/integration/test_webhook.py:44]: `client` and `project` could be mocks, the exceptions are thrown before these are used.

Comment (sfendell-labelbox) [reply tests/integration/test_webhook.py:44]: Good point. I moved them to unit.

Comment (vbrodsky) [reply tests/integration/test_webhook.py:44]: Second Klaus comments, after that can approve

Comment (sfendell-labelbox) [reply tests/integration/test_webhook.py:44]: It should be fixed now and checks are passing, can I get an approval?

**Review by kopreschko — APPROVED** [review]
nice work Sam!

**Pull request merged**